### ISSUE TITLE
Appending slices unreliable behaviour.

### ISSUE THREAD

**Issue by Armatorix** [golang/go#29710]
### What version of Go are you using (`go version`)?
1.11.1

### What did you do?
I've append two slices. 
https://play.golang.org/

### What did you expect to see?
```
ID: [0 1 2 3 4 5 6 7 8 9]
IDX: [0 1 0 1 2 3 4 5 6 7 8 9]
ID: [0 1 5 6 7 8 9 7 8 9]
IDX: [0 1 5 6 7 8 9]
```

### What did you see instead?
```
ID: [0 1 2 3 4 5 6 7 8 9]
IDX: [0 1 0 1 2 3 4 5 6 7 8 9]
ID: [0 1 2 3 4 5 6 7 8 9]
IDX: [0 1 5 6 7 8 9]
```

### what is the purpose?

So IMO this function is not reliable, because when there is enough space in the allocated memory we're changing arguments memory

**Comment by tomor** [comment]
IMO the method behaves exactly as described in the docs: https://golang.org/pkg/builtin/#append

**Comment by Armatorix** [comment]
Yes, it is working as described. But I'm wondering if it is a good approach. IMO `append` loses its semantic meaning when working like this.

**Comment by ianlancetaylor** [comment]
Slices work as they are documented.  That is not something we are going to change at this point in the language's development.

**Comment by ianlancetaylor** [comment]
The language acts as is documented and intended.  I understand that you find it surprising, but there are good reasons for it to act this way, and more importantly, as I said, this is not something that can be changed at this point in the language's development.  If you need to discuss this further, please use a forum; see https://golang.org/wiki/Questions .

**Comment by Armatorix** [comment]
Thx for the link, it will be probably better to use other discussion channel.
